tests/custom_lookups/tests.py

```diff
@@ -647,3 +657,19 @@ def test_subquery_usage(self):
                 id__in=Author.objects.filter(age__div3=2)
             )
             self.assertSequenceEqual(qs, [a2])
+
+class InstanceLookupTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.obj1 = Company.objects.create(name='apple', place='america')
+        cls.obj1 = Company.objects.create(name='banana', place='bangladesh')
+        cls.obj1 = Company.objects.create(name='cherry', place='china')
+        cls.obj1 = Company.objects.create(name='dates', place='denmark')
+
+    def test_register_instance_lookup(self):
+        Company.name.field.register_lookup(StartsWith)
+        self.assertEqual(Company.name.field.instance_lookups, {"sw":StartsWith})
+
+    def test_unregister_instance_lookup(self):
+        Company.name.field._unregister_lookup(StartsWith)
+        self.assertEqual(Company.name.field.instance_lookups, {})
```

Comment:
The current tests are checking only a implementation details not that it actually works. We should add a test with all three layers (class, instance, field) and check that different lookups are used.

Reply:
Yeah. I think those working tests could be added after implementing `get_lookups` methods (would be added soon). These tests just check if the methods work as they should.